include/rocksdb/filter_policy.h

```diff
@@ -65,6 +65,32 @@ class FilterBitsBuilder {
   // The ownership of actual data is set to buf
   virtual Slice Finish(std::unique_ptr<const char[]>* buf) = 0;
 
+  // Similar to Finish(std::unique_ptr<const char[]>* buf), except that
+  // for a non-null status pointer argument, it should point to
+  // Status::Corruption() when there is any corruption during filter
+  // construction or Status::OK() otherwise.
+  virtual Slice Finish(std::unique_ptr<const char[]>* buf,
```

Comment:
Note: I decided to add a status output argument in `Finish` to communicate corruption status. Another solution is to rely on the length of filter content being 0, which might collide with the `hash_entries_.empty() == true` [case](https://github.com/facebook/rocksdb/blob/6.27.fb/table/block_based/filter_policy.cc#L508-L512) in Ribbon filter. It is also less understandable and less standard in my opinion.

Downside of the current approach is name collision of two Finish function could trigger `-Werror=overloaded-virtual`. We could change the function name to `FinishWithStatus` like `[GetBuilderWithContext](https://github.com/facebook/rocksdb/blob/6.27.fb/include/rocksdb/filter_policy.h#L203-L208)` to avoid this. But considering it might be an infrequent case to customize FilterBitsBuilder since it's almost expert work to do so, we might be able to also leave this consequence as it is.